setup.py

```diff
@@ -226,41 +185,44 @@ def setup_package():
 
         metadata['version'] = VERSION
     else:
-        numpy_status = get_numpy_status()
-        numpy_req_str = "scikit-learn requires NumPy >= {0}.\n".format(
-            NUMPY_MIN_VERSION)
-        scipy_status = get_scipy_status()
-        scipy_req_str = "scikit-learn requires SciPy >= {0}.\n".format(
-            SCIPY_MIN_VERSION)
-
-        instructions = ("Installation instructions are available on the "
-                        "scikit-learn website: "
-                        "http://scikit-learn.org/stable/install.html\n")
-
-        if numpy_status['up_to_date'] is False:
-            if numpy_status['version']:
-                raise ImportError("Your installation of Numerical Python "
-                                  "(NumPy) {0} is out-of-date.\n{1}{2}"
-                                  .format(numpy_status['version'],
-                                          numpy_req_str, instructions))
-            else:
-                raise ImportError("Numerical Python (NumPy) is not "
-                                  "installed.\n{0}{1}"
-                                  .format(numpy_req_str, instructions))
-        if scipy_status['up_to_date'] is False:
-            if scipy_status['version']:
-                raise ImportError("Your installation of Scientific Python "
-                                  "(SciPy) {0} is out-of-date.\n{1}{2}"
-                                  .format(scipy_status['version'],
-                                          scipy_req_str, instructions))
-            else:
-                raise ImportError("Scientific Python (SciPy) is not "
-                                  "installed.\n{0}{1}"
-                                  .format(scipy_req_str, instructions))
-
-        from numpy.distutils.core import setup
-
-        metadata['configuration'] = configuration
+        try:
+            metadata['configuration'] = configuration
+            from numpy.distutils.core import setup
+
+            metadata['configuration'] = configuration
+        except ImportError:
+            try:
+                from setuptools import setup
+            except ImportError:
+                from distutils.core import setup
+
+    build_requires = []
+    try:
+        import numpy  # noqa: F401
+    except ImportError:  # We do not have numpy installed
+        build_requires += ['numpy>={0}'.format(NUMPY_MIN_VERSION)]
+    else:
+        # If we're building a wheel, assume there already exist numpy wheels
+        # for this platform, so it is safe to add numpy to build requirements.
+        build_requires += (
+            ['numpy>={0}'.format(NUMPY_MIN_VERSION)]
+            if 'bdist_wheel' in sys.argv[1:] else [])
```

Comment:
I think this would be clearer as an if statement rather than appending an empty list.

Also, in the case where we add [], should we be checking that the numpy version meets minimum requirements?